crates/hir-def/src/dyn_map.rs

```diff
@@ -112,6 +112,10 @@ pub struct Key<K, V, P = (K, V)> {
 }
 
 impl<K, V, P> Key<K, V, P> {
+    #[allow(
+        clippy::new_without_default,
+        reason = "this a const fn, so it can't be default yet. See <https://github.com/rust-lang/rust/issues/63065>"
+    )]
```

Comment:
Seeing this PR I was wondering about whether they finally fixed this or not. Guess they haven't ... (as in that the lint is still firing incorrectly here)